lib/metrics/StatsClient.js

```diff
@@ -105,6 +105,49 @@ class StatsClient {
         return this._redis.incrEx(key, this._expiry, callback);
     }
 
+    /**
+    * wrapper on `getStats` that handles a list of keys
+    * @param {object} log - Werelogs request logger
+    * @param {array} ids - service identifiers
+    * @param {callback} cb - callback to call with the err/result
+    * @return {undefined}
+    */
+    getAllStats(log, ids, cb) {
+        if (!this._redis) {
+            return cb(null, {});
+        }
+
+        const statsRes = {
+            'requests': 0,
+            '500s': 0,
+            'sampleDuration': this._expiry,
+        };
+        let requests = 0;
+        let errors = 0;
+
+        return async.each(ids, (id, done) => {
```

Comment:
Wondering if there should be a limit imposed on the number of connections to the redis DB. Perhaps using `eachLimit` with a concurrency of something like 10 would be safer.

Reply:
good call, will push up this change soon